python/ray/workflow/api.py

```diff
@@ -636,6 +637,39 @@ def continuation(dag_node: "DAGNode") -> Union[Workflow, ray.ObjectRef]:
     return ray.get(dag_node.execute())
 
 
+@PublicAPI(stability="beta")
+def options(
+    **workflow_options: Dict[str, Any]
+) -> Callable[[RemoteFunction], RemoteFunction]:
+    # TODO(suquark): More rigid arguments check like @ray.remote arguments. This is
+    # fairly complex, but we should enable it later.
+    valid_options = {
+        "name",
+        "metadata",
+        "catch_exceptions",
+        "max_retries",
```

Comment:
Shouldn't max retries be unified?

Reply:
I am not sure if it can be/should be unified. The max_retries in workflow retries at the python program layer and does not catch failures beyond normal exceptions. @iycheng

Reply:
The semantics is different. I'd say either leave two APIs here or we give ray's max retries some default value.